src/content_script.js

```diff
@@ -32,50 +80,411 @@ function itemWidthCheck (target) {
+  if (!target || !(typeof target === "object")) {
+    target = document.querySelector("." + target);
+  }
+
+  checkVisibilityAndAppleClass(target, htmlBadgeDiv);
+
+  if (typeof smallSwitch === "undefined") {
+    if (htmlBadgeDiv.classList.contains("fbc-badge-small")) {
+      smallSwitch = true;
+    }
+  }
+
+  // Set offset size based on large/small badge
+  const [elementSizeOffsetX, elementSizeOffsetY] = elementSizeOffsetXY(smallSwitch);
+
+  // Define target element width
+  if (!targetWidth) {
+    targetWidth = parseInt(target.offsetWidth, 10);
+  }
+
+  // Get position coordinates
+  const bodyRect = document.body.getBoundingClientRect();
+  const elemRect = target.getBoundingClientRect();
+
+  // Determine if target element is fixed, will resets or applies class and set appor offset.
+  const {offsetPosX, offsetPosY} = getOffsetsAndApplyClass(elemRect, bodyRect, target, htmlBadgeDiv);
+
+  const htmlBadgeDivPosX = (offsetPosX + targetWidth) - elementSizeOffsetX;
+  const htmlBadgeDivPosY = offsetPosY - elementSizeOffsetY;
+
+  // TODO: Add Zindex Targeting
+  // const targetZindex = calcZindex(target);
+
+  // Set badge position based on target coordinates/size
+  // htmlBadgeDiv.style.zIndex = targetZindex;
+  htmlBadgeDiv.style.left = htmlBadgeDivPosX + "px";
+  htmlBadgeDiv.style.top = htmlBadgeDivPosY + "px";
+}
+
+// List of badge-able in-page elements
+const facebookDetectedElementsArr = [];
+
+function detectFacebookOnPage () {
+  if (!checkForTrackers) {
+    return;
+  }
+  for (let querySelector of SHARE_PATTERN_DETECTION_SELECTORS) {
     for (let item of document.querySelectorAll(querySelector)) {
       // overlay the FBC icon badge on the item
-      if (!item.classList.contains("fbc-overlay")) {
-        item.classList.add(...itemWidthCheck(item));
+      if (!item.classList.contains("fbc-has-badge")) {
+        const itemUIDClassName = "fbc-UID_" + (facebookDetectedElementsArr.length + 1);
+        const itemUIDClassTarget = "js-" + itemUIDClassName;
+        const socialAction = "share";
+        facebookDetectedElementsArr.push(itemUIDClassName);
+        addFacebookBadge(item, itemUIDClassTarget, socialAction);
+        item.classList.add("fbc-has-badge");
+        item.classList.add(itemUIDClassName);
       }
+    }
+  }
+  for (let querySelector of LOGIN_PATTERN_DETECTION_SELECTORS) {
+    for (let item of document.querySelectorAll(querySelector)) {
+      // overlay the FBC icon badge on the item
+      if (!item.classList.contains("fbc-has-badge")) {
+        const itemUIDClassName = "fbc-UID_" + (facebookDetectedElementsArr.length + 1);
+        const itemUIDClassTarget = "js-" + itemUIDClassName;
+        const socialAction = "login";
+        facebookDetectedElementsArr.push(itemUIDClassName);
+        addFacebookBadge(item, itemUIDClassTarget, socialAction);
+        item.classList.add("fbc-has-badge");
+        item.classList.add(itemUIDClassName);
+      }
+    }
+  }
```

Comment:
Lots of code duplication and nested `for` loops here. Should be refactored into a single function that can accept `socialAction` as parameter.